### PR TITLE
Release/2.5.0

### PR DESCRIPTION
**Pull request by 200sc**
Release notes pending

Notes:
- Debug Console:
-- Adds `DisableDebugConsole` configuration option on startup.
-- Adds the `help` built in debug command to print possible commands.
-- Adds `oak.RunCommand()` to simulate a debug console instruction.

- Event Bus:
-- Adds `EventRefreshRate` configuration option on startup.
-- The `event.Bus` type now supports being given a specific refresh rate, instead of having busy waiting on any pending bind or unbind requests. 
-- `event.Bus.ResolvePending()` had a goroutine leak, and this has been fixed.

- Batch Loading:
-- Batch loading can now be configured to replace large images and audio files with blank versions, for faster test iteration in development.
-- Batch loading is now parallelized. 

- Buttons:
-- Adds additional text helpers for `entities/x/btn` constructors. 
-- Adds support for cutting buttons into shapes. 

- Misc:
-- Adds `render/mod.CutShape()`. 
-- Adds `oak.SetTitle()` to change the running window title.
-- Adds `audio.Audio.SetVolume` to change a running audio's underlying volume level. 
-- `render.Text.ToSprite()` has additional safety. Sometimes if a font or text changes in the middle of a `ToSprite` call it could panic deep inside of our font dependency. The font is now copied ahead of time to reduce this risk. 

Tasks before release:
~1. Write release notes~
2. Review test coverage
3. Code review 
4. Should this be 2.5 or 2.4.1? It's a mixed bag of bugfixes and new features.

Related issues:
#145 
#147

### CODE REVIEW

File: audio/load.go [hunk 1]
```diff
@@ -49,70 +50,128 @@ func Get(filename string) (Data, error) {
...
 
 // Unload removes an element from the loaded map. If the element does not
 // exist, it does nothing.
 func Unload(filename string) {
+	loadedLock.Lock()
 	delete(loaded, filename)
+	loadedLock.Unlock()
 }
 
 // IsLoaded is shorthand for (if _, ok := loaded[filename]; ok)
```
Comment (Implausiblyfun) [inline audio/load.go:87]: Given the lock usage perhaps this comment should be updated?

File: config.go [hunk 2]
```diff
@@ -46,23 +51,28 @@ var (
...
+	FrameRate           int              `json:"frameRate"`
+	DrawFrameRate       int              `json:"drawFrameRate"`
+	Language            string           `json:"language"`
+	Title               string           `json:"title"`
+	EventRefreshRate    time.Duration    `json:"refreshRate"`
+	BatchLoad           bool             `json:"batchLoad"`
+	GestureSupport      bool             `json:"gestureSupport"`
+	LoadBuiltinCommands bool             `json:"loadBuiltinCommands"`
+	TrackInputChanges   bool             `json:"trackInputChanges"`
+	DisableDebugConsole bool             `json:"disableDebugConsole"`
```
Comment (Implausiblyfun) [inline config.go:75]: Not a note for this pull but a thought for oak 3.0.
Consider updating json naming to be non-camel but rather _ driven 
ie track_input_changes instead in the tag.

Not a big deal though.

File: audio/load.go (hunk 1 above)
Comment (200sc) [reply audio/load.go:87]: Discussed offline-- Don't think the lock itself is significant enough to change the comment, the presence of the lock around the map should be transparent to the user

**Comment by 200sc** [comment]
For future reference, the weird codecov `- Hits         5010     3919    -1091     ` delta is because something (Go or codecov) stopped counting function / method signature lines as lines covered. We didn't remove 1100 lines of code.

**Comment by 200sc** [comment]
License Compliance manually resolved.

**Pull request merged**